packages/core/types/@chainlink/index.d.ts

```diff
@@ -102,6 +102,48 @@ declare module '@chainlink/types' {
     warning?: string
   }
 
+
+  export type MakeWSHandler = () => WSSubscriptionHandler
+
+  export interface WSSubscriptionHandler {
+    /**
+     * Connection information
+     */
+    connection: {
+      url: string
+      protocol?: any
+    }
+    /**
+     * Gets the subscription message necessary to subscribe to the feed channel
+     */
+    subscribe: (input: AdapterRequest) => any
+    /**
+     * Gets the unsubscription message necessary to unsubscribe to the feed channel
+     */
+    unsubscribe: (input: any) => any
```

Comment:
What if the WS subscription is not supported for this input type:

```suggestion
    unsubscribe: (input: any) => any | undefined
```